### PR TITLE
 Bug 1355636 - Create Deeplinks for MMA

### PR DESCRIPTION
**Pull request by bkmunar**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
AppDelegate.swift | 38 | Force casts should be avoided. 
AppDelegate.swift | 202 | Force casts should be avoided. 
AppDelegate.swift | 211 | if,for,while,do statements shouldn't wrap their conditionals in parentheses. 
AppDelegate.swift | 244 | if,for,while,do statements shouldn't wrap their conditionals in parentheses. 
AppDelegate.swift | 204 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. 
AppDelegate.swift | 226 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. 
AppDelegate.swift | 141 | When registing for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. 
AppDelegate.swift | 148 | When registing for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by bkmunar** [comment]
@Sdaswani @st3fan for you to review!

File: Client/Application/AppDelegate.swift [hunk 1]
```diff
@@ -359,57 +359,73 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
...
-            case "deep-link":
-                guard let value = item.value else { break }
-                Router.shared.routeURL(value)
-                return true
-            default: ()
+        switch host {
+        case "open-url":
+            var url: String?
+            var isPrivate: Bool = false
+            for item in (components.queryItems ?? []) as [URLQueryItem] {
```
Comment (st3fan) [inline Client/Application/AppDelegate.swift]: The `fxa-signin` case below uses `url.getQuery()` .. isn't that nicer? Then you can just do:

```
if let url = query["url"], let private = query["private"] ?? false {
   ...
}
```

File: Client/Application/AppDelegate.swift [hunk 2]
```diff
@@ -359,57 +359,73 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
...
+                // Ex. firefox://fxa-signin?signin=<token>&someQuery=<data>...
+                if query["signin"] != nil {
+                    let fxaParams: FxALaunchParams
+                    fxaParams = FxALaunchParams(query: query)
+                    launchFxAFromURL(fxaParams)
+                    return true
+                }
+            }
+            break
+        default: ()
```
Comment (st3fan) [inline Client/Application/AppDelegate.swift]: What is this `()` ?

Comment (bkmunar) [inline Client/Application/AppDelegate.swift]: I just carried over from the previous switch statements -- I think it acts as an empty block

**Comment by bkmunar** [comment]
@st3fan back to you

**Pull request merged**